src/sentry/plugins/base/manager.py

```diff
@@ -29,9 +29,13 @@ def plugin_that_can_be_configured(self):
                 yield plugin
 
     def configurable_for_project(self, project, version=1):
+        from sentry.plugins.sentry_mail.models import MailPlugin
+
         for plugin in self.all(version=version):
             if not safe_execute(plugin.can_configure_for_project, project, _with_transaction=False):
```

Comment:
@jeffkwoh The right way to do this is by adding a `can_configure_for_project` method to `MailPlugin` which will handle the logic you want to add.